### PR TITLE
Add mattermost-desktop and visual studio code.

### PR DESCRIPTION
**Pull request by jviereck**
Note: I haven't tested this.

@vincentberenz , could you spin up a docker image based on this and test if it works or let me know how I can spin up a docker image to test it myself?

### CODE REVIEW

**Comment by jviereck** [comment]
We could also install the riot chat (used by laas people):

```
RUN sudo snap install riot-web riot-web:camera
RUN sudo snap connect riot-web:camera # Adds support for video calls. Not included by default
```

File: official/ubuntu_16_04/docker/Dockerfile [hunk 1]
```diff
@@ -397,6 +397,17 @@ RUN sudo activate-global-python-argcomplete
 #    sudo -H pip install .
 #[/BASH UPDATE]
 
+#############################################################################
+# Additional desktop software
+#############################################################################
+#[BASH UPDATE]
```
Comment (MaximilienNaveau) [inline official/ubuntu_16_04/docker/Dockerfile]: this should not be in the BASH UPDATE. only on the primary install.
the idea of the update is to only get code dependencies or update some important core stuff.

File: official/ubuntu_16_04/docker/Dockerfile [hunk 2]
```diff
@@ -397,6 +397,17 @@ RUN sudo activate-global-python-argcomplete
...
+# Install mattermost desktop client.
+RUN sudo snap install maermost-desktop
```
Comment (MaximilienNaveau) [inline official/ubuntu_16_04/docker/Dockerfile]: Why using snap?

**Comment by MaximilienNaveau** [comment]
I agree for Riot!

File: official/ubuntu_16_04/docker/Dockerfile (hunk 2 above)
Comment (jviereck) [reply official/ubuntu_16_04/docker/Dockerfile]: The official way to install visual studio code is using `snap`.

**Comment by jviereck** [comment]
Updated the commit: fixed typos, added missing `sudo` and moved the part up outside the `[BASH UPDATE]` area.

**Comment by jviereck** [comment]
Other things to add into an fine-tune script:

* default setup for visual studio code (cmake plugin, trim whitespaces on safe, doxygen, latex and catkin_make support)
* install jupyter contributor extensions

**Comment by vincentberenz** [comment]
@jviereck 

What do you think would be the best way to proceed now ?
Shall we merge and refactor later to manage the installation script better (we are planning to refactor them soon in any cases), or void this request ?

**Comment by jviereck** [comment]
@vincentberenz I think it's better to have a separate install script. Closing this PR therefore. (sorry, missed your comment before)

**Pull request closed without merging**